### PR TITLE
editoast : add delete endpoint for scenario

### LINKED ISSUE #3599: Migrate delete endpoint for Scenario
## Description and goal

Migrate delete endpoint for project from api to editoast in order to migrate API in Editoast. 

## Acceptance criteria

- [ ] Update openapi 
- [ ] Mark the endpoint in Api's openapi as deprectated
- [ ] Write test
- [ ] The endpoint should work

### PR DESCRIPTION
**Pull request by younesschrifi**
Close #3599

### CODE REVIEW

File: editoast/src/models/scenario.rs [hunk 1]
```diff
@@ -57,3 +62,106 @@ pub struct ScenarioWithTrains {
     #[diesel(sql_type = Array<TrainSchedule>)]
     pub train_schedules: Vec<TrainSchedule>,
 }
+
+impl Delete for Scenario {
+    fn delete_conn(conn: &mut diesel::PgConnection, scenario_id: i64) -> Result<bool> {
```
Comment (flomonster) [inline editoast/src/models/scenario.rs:118]: We can add documentation?

File: editoast/src/models/scenario.rs [hunk 2]
```diff
@@ -57,3 +62,106 @@ pub struct ScenarioWithTrains {
...
+        // Delete timetable
+        match delete(
+            timetable_dsl::osrd_infra_timetable
+                .filter(timetable_dsl::id.eq(scenario.timetable.unwrap())),
+        )
+        .execute(conn)
+        {
+            Ok(_) => Ok(true),
+            Err(e) => Err(e.into()),
+        }
```
Comment (flomonster) [inline editoast/src/models/scenario.rs]: The timetable is not deleted in cascade?

File: editoast/src/views/scenario.rs [hunk 3]
```diff
@@ -23,11 +24,15 @@ enum ScenarioError {
...
+    Project { project_id: i64 },
     /// Couldn't found the study with the given study_id
     #[error("Study '{study_id}', could not be found")]
     #[editoast_error(status = 404)]
-    StudyNotFound { study_id: i64 },
+    Study { study_id: i64 },
+    /// Couldn't found the scenario with the given study_id
+    #[error("Study '{scenario_id}', could not be found")]
+    #[editoast_error(status = 404)]
+    Scenario { scenario_id: i64 },
```
Comment (flomonster) [inline editoast/src/views/scenario.rs]: Keeping the `NotFound` suffix is useful to have a better error id.

File: editoast/src/views/scenario.rs [hunk 4]
```diff
@@ -105,3 +111,104 @@ async fn create(
...
+        return Err(ScenarioError::Scenario { scenario_id }.into());
+    }
+
+    // Update project last_modification field
+    let project = project.update_last_modified(db_pool.clone()).await?;
+    project.expect("Project should exist");
+
+    // Update study last_modification field
+    let study = study.update_last_modified(db_pool).await?;
+    study.expect("Study should exist");
```
Comment (flomonster) [inline editoast/src/views/scenario.rs:138]: We could factorize this code with a private utility function.

**Review by flomonster — CHANGES_REQUESTED** [review]
Thanks for this PR. I have a question and a suggestion.

File: editoast/src/models/scenario.rs (hunk 2 above)
Comment (younesschrifi) [reply editoast/src/models/scenario.rs]: No the timetable is not deleted in cascade. We have the same issue as for image in project

File: editoast/src/models/scenario.rs (hunk 1 above)
Comment (younesschrifi) [reply editoast/src/models/scenario.rs:118]: Of course, thanks !

File: editoast/src/views/scenario.rs (hunk 3 above)
Comment (younesschrifi) [reply editoast/src/views/scenario.rs]: There are clippy warnings when I keep NotFound, what should I do ?

File: editoast/src/models/scenario.rs (hunk 2 above)
Comment (flomonster) [reply editoast/src/models/scenario.rs]: Okay, let's keep it that way. The foreign key can be reversed later.

File: editoast/src/models/scenario.rs [hunk 5]
```diff
@@ -57,3 +62,108 @@ pub struct ScenarioWithTrains {
...
```
Comment (flomonster) [inline editoast/src/models/scenario.rs]:
```suggestion
        delete(
            timetable_dsl::osrd_infra_timetable
                .filter(timetable_dsl::id.eq(scenario.timetable.unwrap())),
        )
        .execute(conn)?;
        Ok(true)
```

File: editoast/src/views/scenario.rs [hunk 6]
```diff
@@ -105,3 +123,98 @@ async fn create(
...
+        )
+    }
+
+    pub fn delete_scenario_request(project_id: i64, study_id: i64, scenario_id: i64) -> Request {
+        TestRequest::delete()
+            .uri(
+                format!("/projects/{project_id}/studies/{study_id}/scenarios/{scenario_id}")
+                    .as_str(),
+            )
+            .to_request()
```
Comment (flomonster) [inline editoast/src/views/scenario.rs]:
```suggestion
        let url = format!("/projects/{project_id}/studies/{study_id}/scenarios/{scenario_id}");
        TestRequest::delete().uri(url.as_str()).to_request()
```

**Review by flomonster — CHANGES_REQUESTED** [review]

File: editoast/src/tables.rs [hunk 7]
```diff
@@ -239,7 +239,7 @@ table! {
 joinable!(osrd_infra_study -> osrd_infra_project (project_id));
 
 table! {
-    osrd_infra_scenario {
+    osrd_infra_scenario(id) {
```
Comment (flomonster) [inline editoast/src/tables.rs]:
```suggestion
    osrd_infra_scenario {
```

**Review by flomonster — CHANGES_REQUESTED** [review]

**Review by flomonster — APPROVED** [review]

**Pull request merged**